espree.js

```diff
@@ -3857,6 +3858,15 @@ function parseParams(firstRestricted) {
                     message = Messages.StrictParamDupe;
                 }
             }
+            if (match("=")) {
+                if (!allowDefaultParams) {
+                    throwUnexpected(lookahead);
+                }
+                lex();
+                def = parseAssignmentExpression();
+                ++defaultCount;
+            }
+            defaults.push(def);
```

Comment:
Is there a reason to always push `def` into `defaults`, even when a `=` isn't matched/found?